registry/remote/repository.go

```diff
@@ -1120,6 +1120,9 @@ func (s *manifestStore) deleteWithIndexing(ctx context.Context, target ocispec.D
 		if err := limitSize(target, s.repo.MaxMetadataBytes); err != nil {
 			return err
 		}
+		// "pull" and "delete" scopes are required, "push" is potentially needed
+		// for client-side indexing
+		ctx = auth.AppendRepositoryScope(ctx, s.repo.Reference, auth.ActionPull, auth.ActionPush, auth.ActionDelete)
```

Comment:
According to the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege), we should not request "push" scope unless it is required.

Reply:
Will the `pull` scope hint help reduce the auth request count if client calls `Exists` before calling `manifestStore.Delete`? Shouldn't the optimization be done on caller side?

Reply:
@shizhMSFT Removed "push" scope request.
@qweeah Yes, and the optimization has already been applied in [`Push`](https://github.com/oras-project/oras-go/blob/main/registry/remote/repository.go#L835), [`Push Reference`](https://github.com/oras-project/oras-go/blob/main/registry/remote/repository.go#L1272) and [`Mount`](https://github.com/oras-project/oras-go/blob/main/registry/remote/repository.go#L757-L762) at the first place.

Reply:
> Will the pull scope hint help reduce the auth request count if client calls Exists before calling manifestStore.Delete

That should be done on the caller side.